#!/usr/bin/env python
PACKAGE = "lidar_align_vis"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()

tuning_range_m = 0.2
tuning_range_deg = 10.0
tuning_range_rad = (tuning_range_deg / 180.0) * math.pi
tuning_range_sec = 0.2;

gen.add("dtx", double_t,    0, "Translation x", 0, -tuning_range_m, tuning_range_m)
gen.add("dty", double_t,    0, "Translation y", 0, -tuning_range_m, tuning_range_m)
gen.add("dtz", double_t,    0, "Translation z", 0, -tuning_range_m, tuning_range_m)
gen.add("drx", double_t,    0, "Translation x", 0, -tuning_range_rad, tuning_range_rad)
gen.add("dry", double_t,    0, "Translation y", 0, -tuning_range_rad, tuning_range_rad)
gen.add("drz", double_t,    0, "Translation z", 0, -tuning_range_rad, tuning_range_rad)
gen.add("d_time_offset", double_t,    0, "Time offset", 0, -tuning_range_sec, tuning_range_sec)
gen.add("reset_to_init",   bool_t,   0, "Reset to initial transform",  False)

exit(gen.generate(PACKAGE, "lidar_align_vis", "Calib"))
